File: definitive_push.py
import subprocess
import os

def run_git_cmd(cmd, log_file):
    with open(log_file, "a") as f:
        f.write(f"\nRunning: {cmd}\n")
    try:
        result = subprocess.run(cmd, shell=True, capture_output=True, text=True)
        with open(log_file, "a") as f:
            f.write(f"STDOUT: {result.stdout}\n")
            f.write(f"STDERR: {result.stderr}\n")
        return result.returncode == 0
    except Exception as e:
        with open(log_file, "a") as f:
            f.write(f"EXCEPTION: {str(e)}\n")
        return False

log_file = "e:/brkthru-digital-landing/push_diagnosis.txt"
if os.path.exists(log_file):
    os.remove(log_file)

os.chdir("e:/brkthru-digital-landing")

print("Logging to push_diagnosis.txt")
run_git_cmd("git status", log_file)
run_git_cmd("git add .", log_file)
run_git_cmd('git commit -m "FINAL: Restored Universal Header site-wide"', log_file)
run_git_cmd("git remote -v", log_file)
run_git_cmd("git push origin master:main --force", log_file)
run_git_cmd("git push origin master:master --force", log_file)

File: definitive_push.js
const { execSync } = require('child_process');

function runGit(cmd) {
    console.log(`Running: ${cmd}`);
    try {
        const stdout = execSync(cmd, { encoding: 'utf8' });
        console.log(`STDOUT: ${stdout}`);
        return true;
    } catch (error) {
        console.log(`ERROR: ${error.message}`);
        console.log(`STDERR: ${error.stderr}`);
        return false;
    }
}

try {
    process.chdir('e:/brkthru-digital-landing');
    console.log(`Current Dir: ${process.cwd()}`);

    runGit('git status');
    runGit('git add .');
    runGit('git commit -m "FINAL: Restored Universal Header site-wide and cleaned duplicates"');
    runGit('git push origin master:main --force');
    runGit('git push origin master:master --force');
} catch (e) {
    console.log(`Global Error: ${e.message}`);
}
